SU2_CFD/src/fluid/CFluidScalar.cpp

```diff
@@ -0,0 +1,245 @@
+#include <cmath>
+#include <math.h>
+#include <vector>
+#include <numeric>
+#include <iostream>
+
+#include "../../include/fluid/CFluidScalar.hpp"
+#include "../../include/fluid/CConstantViscosity.hpp"
+#include "../../include/fluid/CSutherland.hpp"
+#include "../../include/fluid/CPolynomialViscosity.hpp"
+#include "../../include/fluid/CConstantConductivity.hpp"
+#include "../../include/fluid/CConstantConductivityRANS.hpp"
+#include "../../include/fluid/CConstantPrandtl.hpp"
+#include "../../include/fluid/CConstantPrandtlRANS.hpp"
+#include "../../include/fluid/CPolynomialConductivity.hpp"
+#include "../../include/fluid/CPolynomialConductivityRANS.hpp"
+#include "../../include/fluid/CIncIdealGas.hpp"
+
+CFluidScalar::CFluidScalar(CConfig *config, const su2double value_pressure_operating) : CFluidModel() {
+  unsigned short n_scalars = config->GetNScalarsInit();
+  config->SetNScalarsInit(n_scalars);
+  n_species_mixture = n_scalars + 1;
+
+  specificHeat.resize(n_species_mixture);
+  molarMasses.resize(n_species_mixture);
+  massFractions.resize(n_species_mixture);
+  moleFractions.resize(n_species_mixture);
+  laminarViscosity.resize(n_species_mixture);
+  laminarthermalConductivity.resize(n_species_mixture);
+
+  for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+    molarMasses.at(iVar) = config->GetMolecular_Weight(iVar);
+    specificHeat.at(iVar) = config->GetSpecific_Heat_Cp(iVar);
+  }
+
+  wilke = false;
+  davidson = true;
+
+  Pressure = value_pressure_operating;
+  Gas_Constant = config->GetGas_Constant();
+  Gamma = 1.0;
+
+  SetLaminarViscosityModel(config);
+  SetThermalConductivityModel(config);
+}
+
+void CFluidScalar::SetLaminarViscosityModel(const CConfig* config) {
+  switch (config->GetKind_ViscosityModel()) {
+    case VISCOSITYMODEL::CONSTANT:
+    /* Build a list of LaminarViscosity pointers to be used in e.g. wilkeViscosity to get the species viscosities. */
+      for (int iVar = 0; iVar < n_species_mixture; iVar++){
+        LaminarViscosityPointers[iVar] = std::unique_ptr<CConstantViscosity>(new CConstantViscosity(config->GetMu_Constant(iVar)));
+      }
+      break;
+    case VISCOSITYMODEL::SUTHERLAND:
+      for (int iVar = 0; iVar < n_species_mixture; iVar++){
+        LaminarViscosityPointers[iVar] = std::unique_ptr<CSutherland>(new CSutherland(config->GetMu_Ref(iVar), config->GetMu_Temperature_Ref(iVar), config->GetMu_S(iVar)));
```

Comment:
is it possible that different species have different viscosity models? 

Currently not, it seems, but in case sth like `config->GetKind_ViscosityModel(>>iVar<<)` shows up we would need to swap the order of switch and for loop.